tests/phpunit/Mutator/Augmentation/ExpressionRepeatTest.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * This code is licensed under the BSD 3-Clause License.
+ *
+ * Copyright (c) 2017, Maks Rafalko
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Tests\Mutator\Augmentation;
+
+use Infection\Tests\Mutator\BaseMutatorTestCase;
+
+final class ExpressionRepeatTest extends BaseMutatorTestCase
+{
+    /**
+     * @dataProvider mutationsProvider
+     *
+     * @param string|string[] $expected
+     */
+    public function test_it_can_mutate(string $input, $expected = []): void
+    {
+        $this->doTest($input, $expected);
+    }
+
+    public function mutationsProvider(): iterable
+    {
+        yield 'It duplicates simple statement' => [
+            <<<'PHP'
+<?php
+
+$a->foo();
+PHP
+            ,
+            <<<'PHP'
+<?php
+
+$a->foo();
+$a->foo();
+PHP
+            ,
+        ];
+
+        yield 'It duplicates complex statement' => [
+            <<<'PHP'
+<?php
+
+$z = $b->bar() + [1 => $c->foo()];
+PHP
+            ,
+            <<<'PHP'
+<?php
+
+$z = $b->bar() + [1 => $c->foo()];
+$z = $b->bar() + [1 => $c->foo()];
```

Comment:
Won't we have many *new* hard-to-kill mutations with mutating *any* expression?

1. Probably it's safer to mutate only method calls with side effects? (see `MethodCallRemoval`)

    for example, mutate this
    ```diff
    $this->foo()
    +$this->foo()
    ```

    but **not** this
    ```diff
    $a = $this->foo() + $this->bar();
    + $a = $this->foo() + $this->bar();
    ```

    Or, probably, default to mutating only side-effects method calls but add a setting for this mutator to mutate *any* expression. Wdyt?

    Now MSI is quite low for such mutator https://monosnap.com/direct/cb98GfVJArUligeK0FLRboh5dQrSa1

3. Also, what about static calls? `Any::foo()` - is it mutated? let's add it to tests, as if it is, this will be extremely hard to kill

Reply:
Good point that is should try very hard to be as useful as it can. Let me see if I can limit it to only method-call containing statements.

Reply:
I updated it to mutate only statements with method calls, but without static or function calls. This way we can be certain a mutation can be tested against. Can you confirm on your end?

As I see it, we get:
```
343 mutations were generated:
     157 mutants were killed
       0 mutants were not covered by tests
     186 covered mutants were not detected
       0 errors were encountered
       0 time outs were encountered

Metrics:
         Mutation Score Indicator (MSI): 45%
         Mutation Code Coverage: 100%
         Covered Code MSI: 45%
```

I'd say we have far worse mutations, which are often almost impossible to test against. 

| Mutator                         | Mutations | Killed | Escaped | Errors | Timed Out | MSI (%s) | Covered MSI (%s) |
| ------------------------------- | --------- | ------ | ------- | ------ | --------- | -------- | ---------------- |
| ExpressionRepeat                |       343 |    162 |     181 |      0 |         0 |    47.23 |            47.23 |
| CastFloat                       |         7 |      3 |       4 |      0 |         0 |    42.86 |            42.86 |
| GreaterThanOrEqualTo            |         7 |      3 |       4 |      0 |         0 |    42.86 |            42.86 |
| RoundingFamily                  |        14 |      6 |       8 |      0 |         0 |    42.86 |            42.86 |
| UnwrapArrayValues               |         5 |      2 |       3 |      0 |         0 |    40.00 |            40.00 |
| UnwrapStrToLower                |         1 |      0 |       1 |      0 |         0 |     0.00 |             0.00 |
| ProtectedVisibility             |         1 |      0 |       1 |      0 |         0 |     0.00 |             0.00 |
| LessThanOrEqualTo               |         1 |      0 |       1 |      0 |         0 |     0.00 |             0.00 |
| CastObject                      |         1 |      0 |       1 |      0 |         0 |     0.00 |             0.00 |